tests/cocotb/basic/test/test_hwpe_stream_split.py

```diff
@@ -0,0 +1,301 @@
+        pop_strb  = [0]*NB_OUT_STREAMS
+        pop_ready = [0]*NB_OUT_STREAMS
+
+        # Set stimuli values to drive DUT
+        push_data  = random.randint(0,MAX_DATA_VAL_I)
+        push_valid = random.randint(0,1)
+        push_strb  = random.randint(0,MAX_STRB_VAL_I)
+        
+        # Load stimuli values to push in
+        dut. data_i.value = push_data
+        dut.valid_i.value = push_valid
+        dut. strb_i.value = push_strb
+
+        for j in range(NB_OUT_STREAMS):
+            pop_ready[j]         = random.randint(0,1)
+            dut.ready_o[j].value = pop_ready[j]
+
+        # Propagate data
+        await Timer(1, units="ns")
+
+        #-----------------------------------
+        # Extracting outputs
+        #-----------------------------------
+        for j in range(NB_OUT_STREAMS):
+
+            pop_data [j] = dut. data_o[j].value
+            pop_valid[j] = dut.valid_o[j].value 
+            pop_strb [j] = dut. strb_o[j].value 
+            pop_ready[j] = dut.ready_o[j].value
+
+        # Push ready direction is an output to the input side
+        push_ready = dut.ready_i.value
+
+
+        #-----------------------------------
+        # Logging data
+        #-----------------------------------
+        cocotb.log.info(f'===== INPUT LOG =====')
+
+        cocotb.log.info(f'----- INPUT STREAM -----')
+        cocotb.log.info(f'push_i.data.value  = { hex(push_data)}')
+        cocotb.log.info(f'push_i.valid.value = {     push_valid}')
+        cocotb.log.info(f'push_i.strb.value  = { bin(push_strb)}')
+        cocotb.log.info(f'push_i.ready.value = {bin(push_ready)}')
+
+        cocotb.log.info(f'===== OUTPUT LOG =====')
+
+        for j in range(NB_OUT_STREAMS):
+
+            cocotb.log.info(f'----- OUTPUT STREAM # {j} -----')
+            cocotb.log.info(f'pop_o{NB_OUT_STREAMS-j}.data.value  = { hex(pop_data[j])}')
+            cocotb.log.info(f'pop_o{NB_OUT_STREAMS-j}.valid.value = {     pop_valid[j]}')
+            cocotb.log.info(f'pop_o{NB_OUT_STREAMS-j}.strb.value  = { bin(pop_strb[j])}')
+            cocotb.log.info(f'pop_o{NB_OUT_STREAMS-j}.ready.value = {bin(pop_ready[j])}')
+
+        #-----------------------------------
+        # Assertion checks
+        #-----------------------------------
+
+        # Checking for output mismatch
+        # Expected output should be a simple concatenation of inputs
+        # Note that MSB is higher number in array
+        data_check  = 0
+        strb_check  = 0
+        valid_check = 0
+        ready_check = 0
+
+        for j in range(NB_OUT_STREAMS-1,-1,-1):
+
+            data_check   = (data_check << DATA_WIDTH_OUT) + pop_data[j]
+            strb_check   = (strb_check << STRB_WIDTH_OUT) + pop_strb[j]
+            valid_check += pop_valid[j]
+            ready_check += pop_ready[j]
+
+        # Check if data is correct
+        assert(data_check == push_data), f"ERROR! Output mismatch - Expected output: {hex(data_check)}; Actual output: {hex(push_data)}"
```

Comment:
You are writing these assertions very verbosely. I think you could just use:
assert(data_check == push_data). This will print an AssertionError in python which is legible enough in my opinion. I think pytest can insert the values inside those assertions automatically. Can you check this?

Reply:
Okay I replayed the reason why I made it verbose. When an error happens it just shows something like:

assert 1463020790750066220 != 1463020790750066220

It's not very readable so I thought doing it with these messages and hex conversions make them more readable.